cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -520,7 +520,9 @@ private void subscribeOrSync(boolean autoRegister) {
         int lastSync = sharedPreferences.getInt(CleverPushPreferences.SUBSCRIPTION_LAST_SYNC, 0);
         int nextSync = lastSync + threeDays;
         String subscriptionId = sharedPreferences.getString(CleverPushPreferences.SUBSCRIPTION_ID, null);
-        if (subscriptionId == null && autoRegister || subscriptionId != null && nextSync < currentTime) {
+        String subscriptionStatus = sharedPreferences.getString(CleverPushPreferences.CLEVERPUSH_STATUS,"");
+
+        if (!subscriptionStatus.equalsIgnoreCase("UNSUBSCRIBED") && subscriptionId == null && autoRegister || subscriptionId != null && nextSync < currentTime) {
```

Comment:
handle boolean

Reply:
Also please make this if condition more readable (maybe by using 1-2 more functions)